internal/services/mssql/mssql_database_resource.go

```diff
@@ -234,6 +236,7 @@ func resourceMsSqlDatabaseCreateUpdate(d *pluginsdk.ResourceData, meta interface
 			SampleName:                       sql.SampleName(d.Get("sample_name").(string)),
 			RequestedBackupStorageRedundancy: sql.RequestedBackupStorageRedundancy(d.Get("storage_account_type").(string)),
 			ZoneRedundant:                    utils.Bool(d.Get("zone_redundant").(bool)),
+			MaintenanceConfigurationID:       utils.String(maintenanceConfigId.ID()),
```

Comment:
`maintenanceConfigId` cannot be `nil` here?

Reply:
`maintenanceConfigId` has at least SQL_Default value (if not explicitly set)